### PR TITLE
feat: add markdown linting and fix errors

### PR DESCRIPTION
**Pull request by olizilla**
Adds remark-lint as a fancy and configurable markdown linter that we can tweak to help us write better specs.

I have fixed markdown errors as I've added rules, including multiple broken links and missing code blocks. The current output of the linter is clean on this branch, and I've enabled it in CI. 

Run `npm test`  to try it out. Example output:

```bash
⨎ npm test

> specs-new@1.0.0 test /Users/oli/Code/filecoin-project/specs
> remark content --quiet

content/algorithms/crypto/randomness.md
  15:39-15:46  warning  Found reference to undefined definition  no-undefined-references  remark-lint
  54:24-54:31  warning  Found reference to undefined definition  no-undefined-references  remark-lint

⚠ 2 warnings
```

License: MIT
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

**Comment by olizilla** [comment]
added a rule to warn when headings increment more than 1 level at a time.

```bash
⨎ npm test

> specs-new@1.0.0 test /Users/oli/Code/filecoin-project/specs
> remark content --quiet

content/algorithms/porep/stacked_drg_circuit.md
  10:1-10:25  warning  Heading levels should increment by one level at a time  heading-increment  remark-lint

content/libraries/drand/_index.md
  25:1-25:29  warning  Heading levels should increment by one level at a time  heading-increment  remark-lint

⚠ 2 warnings
```

**Comment by olizilla** [comment]
ok this is starting to feel nice...

added a rule to warn when code fences do not have a language flag

```bash
⨎ npm test

> specs-new@1.0.0 test /Users/oli/Code/filecoin-project/specs
> remark content --quiet

content/algorithms/crypto/signatures.md
    64:1-66:4  warning  Missing code language flag  fenced-code-flag  remark-lint

content/algorithms/porep/stacked_drg.md
  255:1-257:4  warning  Missing code language flag  fenced-code-flag  remark-lint
  260:1-262:4  warning  Missing code language flag  fenced-code-flag  remark-lint

content/algorithms/post/election_post.md
  213:1-221:4  warning  Missing code language flag  fenced-code-flag  remark-lint

content/libraries/drand/_index.md
    29:1-35:4  warning  Missing code language flag  fenced-code-flag  remark-lint

⚠ 5 warnings
```

**Comment by olizilla** [comment]
Added rule to ensure all docs start with an h1

```bash
⨎ npm test

> specs-new@1.0.0 test /Users/oli/Code/filecoin-project/specs
> remark content --quiet

content/implementations/lotus.md
  17:1-17:9  warning  First heading level should be `1`  first-heading-level  remark-lint

⚠ 1 warning
```

**Comment by olizilla** [comment]
made a custom lint rule to warn aginst our pattern of putting a horizontal rule after a heading... to fix #1081 

see: https://github.com/olizilla/remark-lint-no-hr-after-heading

```bash
content/systems/filecoin_vm/sysactors/cron_actor.md
     11:1-11:4  warning  Don’t use a horizontal line after a heading  no-hr-after-heading            remark-lint

content/systems/filecoin_vm/sysactors/init_actor.md
     11:1-11:4  warning  Don’t use a horizontal line after a heading  no-hr-after-heading            remark-lint

content/systems/filecoin_vm/sysactors/reward_actor.md
     10:1-10:4  warning  Don’t use a horizontal line after a heading  no-hr-after-heading            remark-lint

⚠ 138 warnings
```

File: .remarkrc.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+plugins:
+  # make remark aware of fontmatter.
+  remark-frontmatter:
+  # sensible linter defaults see: https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-recommended#rules
```
Comment (olizilla) [inline .remarkrc.yaml:4]: This is where we configure what markdown linting rules we want. we start with a preset with a bunch of rules enabled, then add some of our own, then remove some of the preset rules that were too noisey

File: content/algorithms/porep/stacked_drg_circuit.md [hunk 2]
```diff
@@ -31,8 +27,6 @@ This circuit proves that given a Merkle root `CommD`, `CommRLast`, and `commRSta
   - otherwise, execute the function
 - **Inclusion path**: Binary representation of the Merkle tree path that must be proven packed into a single `Fr` element.
 
-# Offline PoRep circuit
```
Comment (olizilla) [inline content/algorithms/porep/stacked_drg_circuit.md:34]: Bring this back says @hugomrdias

**Review by hugomrdias — APPROVED** [review]

**Comment by olizilla** [comment]
@hugomrdias fixed merge conflict and the drg header nesting, and added a note about `npm test` markdown linting to the readme.

Good to merge?

**Review by hugomrdias — APPROVED** [review]

**Pull request merged**